doc/source.rst

```diff
@@ -181,12 +181,16 @@ File patterns
 File path patterns are used for include and omit, and for combining path
 remapping.  They follow common shell syntax:
 
-- ``*`` matches any number of file name characters, not including the directory
-  separator.
-
 - ``?`` matches a single file name character.
 
-- ``**`` matches any number of nested directory names, including none.
+- ``*`` as part of a path component matches any number of file name characters,
```

Comment:
Maybe, we could clarify the distinction between "part of" and "the entire match pattern component" through a concrete example?